components/BottomNavigation/src/MDCBottomNavigationBar.h

```diff
@@ -0,0 +1,105 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+/** States used to configure bottom navigation on when to show item titles. */
+typedef NS_ENUM(NSInteger, MDCBottomNavigationBarTitleVisibility) {
+
+  // Default behavior is to show title when item is selected, hide otherwise.
+  MDCBottomNavigationBarTitleVisibilitySelected = 0,
+
+  // Item titles are always visible.
+  MDCBottomNavigationBarTitleVisibilityAlways = 1,
+
+  // Item titles are never visible.
+  MDCBottomNavigationBarTitleVisibilityNever = 2
+};
+
+/**
+ States used to configure bottom navigation in landscape mode with respect to how items are spaced
+ and item title orientation. Titles will be shown or hidden depending on title hide state.
+ */
+typedef NS_ENUM(NSInteger, MDCBottomNavigationBarDistribution) {
+
+  // Items are distributed using the entire width of the device. Titles are centered below icons.
+  MDCBottomNavigationBarDistributionEqual = 0,
+
+  // Items are distributed using the entire width of the device. Titles are positioned adjacent to
+  // icons.
+  MDCBottomNavigationBarDistributionEqualAdjacentTitles = 1,
+
+  // Items are tightly clustered together. Titles are positioned below icons.
+  MDCBottomNavigationBarDistributionCluster = 2
+};
```

Comment:
super-nit / consider:
MDCBottomNavigatioBarAlignment {
MDCBottomNavigatioBarAlignmentJustified
MDCBottomNavigatioBarAlignmentCentered
MDCBottomNavigatioBarAlignmentCenteredWithAdjacentTitles
}

.... naming is hard.